scalanet/test/src/io/iohk/scalanet/peergroup/InMemoryPeerGroupSpec.scala

```diff
@@ -0,0 +1,117 @@
+package io.iohk.scalanet.peergroup
+
+import io.iohk.scalanet.TaskValues._
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{BeforeAndAfter, FlatSpec}
+import org.scalatest.Matchers._
+import org.scalatest.concurrent.ScalaFutures._
+
+import scala.concurrent.Future
+import scala.util.Random
+
+class InMemoryPeerGroupSpec extends FlatSpec with BeforeAndAfter {
```

Comment:
It looks to me like we could merge most *PeerGroupSpec and test all existing implementations.

Reply:
that is the idea of another ticket and also why I played with the creation of the PeerUtils typeclass